lib/byron/test/unit/Cardano/Wallet/Byron/TransactionSpec.hs

```diff
@@ -159,6 +223,197 @@ spec = do
             , (xprv "address-number-25", Coin 1)
             ] goldenTestnet__1_25
 
+{-------------------------------------------------------------------------------
+                                Properties
+-------------------------------------------------------------------------------}
+
+
+propSizeEstimation
+    :: forall n k.
+       ( PaymentAddress n k
+       , WalletKey k
+       , MaxSizeOf Address n k
+       )
+    => ProtocolMagic
+    -> Gen CoinSelection
+    -> (Int -> Gen [Address])
+    -> Property
+propSizeEstimation pm genSel genChngAddrs =
+    withMaxSuccess 1000 $
+    forAllShrinkBlind genSel shrinkSelection $ \sel -> let nc = length (change sel) in
+    forAllBlind (genChngAddrs nc) $ \chngAddrs ->
```

Comment:
OK so the forAll . forAll is because the generators are parameters of this property.

Reply:
Yeah, the issue here is that generators depends on which network discrimination and the type of address / key that is chosen. Thus in practice, we can't just have `Arbitrary Address`, because there are 4 types of `Address` in this test. We could define arbitrary instances on newtypes that wraps an `Address` with a `Proxy` or alike, but I found that passing the generator as parameters and using forAll was just simpler.